schemes/musica/tuvx/musica_ccpp_tuvx.F90

```diff
@@ -469,18 +475,25 @@ subroutine tuvx_run(temperature, dry_air_density,                  &
     if (errcode /= 0) return
 
     do i_col = 1, size(temperature, dim=1)
-      call calculate_heights( geopotential_height_wrt_surface_at_midpoint(i_col,:),  &
-                              geopotential_height_wrt_surface_at_interface(i_col,:), &
-                              surface_geopotential(i_col),                           &
-                              reciprocal_of_gravitational_acceleration,              &
-                              height_midpoints, height_interfaces )
-      call set_height_grid_values( height_grid, height_midpoints, height_interfaces, &
+
+      ! check if solar zenith angle is within the range to calculate photolysis rate constants
+      solar_zenith_angle_degrees = solar_zenith_angle(i_col) * 180.0_kind_phys / PI
+      if (solar_zenith_angle_degrees > MAX_SOLAR_ZENITH_ANGLE .or. &
+          solar_zenith_angle_degrees < MIN_SOLAR_ZENITH_ANGLE) then
+        photolysis_rate_constants(:,:) = 0.0_kind_phys
+      else
+        call calculate_heights( geopotential_height_wrt_surface_at_midpoint(i_col,:),  &
+                                geopotential_height_wrt_surface_at_interface(i_col,:), &
+                                surface_geopotential(i_col),                           &
+                                reciprocal_of_gravitational_acceleration,              &
+                                 height_midpoints, height_interfaces )
+        call set_height_grid_values( height_grid, height_midpoints, height_interfaces, &
                                    errmsg, errcode )
-      if (errcode /= 0) return
+        if (errcode /= 0) return
 
-      call set_temperature_values( temperature_profile, temperature(i_col,:), &
+        call set_temperature_values( temperature_profile, temperature(i_col,:), &
                                    surface_temperature(i_col), errmsg, errcode )
-      if (errcode /= 0) return
+        if (errcode /= 0) return
 
       call set_cloud_optics_values( cloud_optics, cloud_area_fraction(i_col,:), &
```

Comment:
Could you add space in the lines starting 498 - 503 to align with the `else` conditional code?

Reply:
added!